cmds/core/ping/ping.go

```diff
@@ -160,32 +160,43 @@ func ping1(net6 bool, host string, i uint64, waitFor time.Duration) (string, err
 	return fmt.Sprintf("%d bytes from %v: icmp_seq=%v, time=%v", amt, host, i, latency), nil
 }
 
-func main() {
-	flag.Parse()
-
-	// options without parameters (right now just: -hV)
-	if flag.NArg() < 1 {
-		optwithoutparam()
-	}
+func ping(host string) error {
 	if *packetSize < 8 {
-		log.Fatalf("packet size too small (must be >= 8): %v", *packetSize)
+		return fmt.Errorf("packet size too small (must be >= 8): %v", *packetSize)
 	}
 
 	interval := time.Duration(*intv)
-	host := flag.Args()[0]
-
+	p := GetPing()
 	// ping needs to run forever, except if '*iter' is not zero
 	waitFor := time.Duration(*wtf) * time.Millisecond
 	var i uint64
 	for i = 1; *iter == 0 || i <= *iter; i++ {
-		msg, err := ping1(*net6, host, i, waitFor)
+		msg, err := p.ping1(*net6, host, i, waitFor)
 		if err != nil {
-			log.Fatalf("ping failed: %v", err)
+			return fmt.Errorf("ping failed: %v", err)
 		}
 		if *audible {
 			msg = "\a" + msg
 		}
 		log.Print(msg)
 		time.Sleep(time.Millisecond * interval)
 	}
+	return nil
+}
+
+func main() {
+	flag.Parse()
+	if *version {
```

Comment:
We need to think about the version option. Either all u-root tools should do this uniformly, or none should. 

history: u-root commands did not have versions in the first years because u-root was designed to ship as source code. Versions seems superfluous. 
More history: versions predate git by about 40 years. The question is, what should version return? The version of u-root from whence it came? The ref of the git tree from whence it was built (I actually like this one)? Or some compound of the two? I think we need to think this through a bit mnore.